examples/contrib/oed/item_response.py

```diff
@@ -0,0 +1,133 @@
+import argparse
+import torch
+from torch.distributions import constraints
+import numpy as np
+
+import pyro
+import pyro.distributions as dist
+from pyro import optim
+from pyro.infer import Trace_ELBO
+from pyro.contrib.oed.eig import vi_ape
+import pyro.contrib.gp as gp
+
+# Number of layers of hierarchy (1 layer- classical linear model)
+layers = 2
+# Parameters controlling the prior variance
+A = 20.
+B = 10.
+beta = torch.tensor(1.)
+
+softplus = torch.nn.functional.softplus
+sigmoid = torch.nn.functional.sigmoid
+
+
+def model(design_tensor, participant_id):
+    """Hierarchical logistic regression model.
+
+    :param torch.tensor design_tensor: a `batch_dims x n x p`
+        tensor giving the features of the `n` responses (e.g.
+        the features of the `n` objects to be given at this time)
+    :param list participant_id: a hierarchical list of the groups
+        that the current participant is assigned to. The first element
+        indexes the highest level group. For instance, a geographical
+        hierarchy list could look like 
+        `["United States", "California", "Santa Barbara"]`
```

Comment:
santa barbara is about 200 miles too far to the south to be welcome in this codebase ; )